File: Packet_Maker/bin/Debug/net7.0/Cs/PacketID.cs
public enum PacketID : Int32
{
    C_REQ_LOGIN=1000,
S_RES_LOGIN=2000,

}

public abstract class Packet
{
    public abstract PacketID GetID();
    public abstract Int32 GetSize();
    public abstract Serialize(Buffer buffer);
    public abstract Deserialize(Buffer buffer);
}
}

File: Packet_Maker/bin/Debug/net7.0/Cpp/PacketID.cpp

enum PacketID : int32_t
{
    C_REQ_LOGIN=1000,
S_RES_LOGIN=2000,

};

class Packet
{
public:
    virtual PacketID GetID()=0;
    virtual int32_t GetSize()=0;
    virtual void Serialize(Buffer& buffer)=0;
    virtual void Deserialize(Buffer& buffer)=0;
};
};